src/routes/profiles.py

```
@@ -1,5 +1,113 @@
+        payload = jwt_manager.decode_access_token(token)
+        token_user_id = payload.get("user_id")
+    except TokenExpiredError:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Token has expired."
+        )
+    except InvalidTokenError:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid token."
+        )
+    if user_id != token_user_id:
+        result_user_group = await db.execute(
+            select(UserGroupModel)
+            .join(UserModel)
+            .where(UserModel.id == token_user_id)
+        )
+        user_group = result_user_group.scalar_one_or_none()
+        if not user_group or user_group.name == UserGroupEnum.USER:
+            raise HTTPException(
+                status_code=status.HTTP_403_FORBIDDEN,
+                detail="You don't have permission to edit this profile."
+            )
+
+    result_user = await db.execute(
+        select(UserModel)
+        .where(UserModel.id == user_id)
+    )
+    user = result_user.scalar_one_or_none()
+    if not user or not user.is_active:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="User not found or not active."
+        )
+
+    result_user_profile = await db.execute(
+        select(UserProfileModel)
+        .where(UserProfileModel.user_id == user_id)
+    )
+    user_profile = result_user_profile.scalar_one_or_none()
+    if user_profile:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="User already has a profile."
+        )
+
+    avatar_file_name = f"avatars/{user.id}_{request_data.avatar.filename}"
+    avatar_bytes = await request_data.avatar.read()
+    try:
+        await s3_client.upload_file(file_name=avatar_file_name, file_data=avatar_bytes)
+        avatar_url = await s3_client.get_file_url(avatar_file_name)
+    except Exception:
```

The avatar upload except-block (line 85) currently swallows all exceptions and returns the required 500 message (which satisfies the spec). For better observability, capture the exception object and log it (or re-raise after logging in non-test env). For example:

```py
except Exception as exc:
    logger.exception("Failed to upload avatar", exc_info=exc)
    raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail="Failed to upload avatar. Please try again later.")
```

This keeps the required client-facing message but preserves server-side diagnostics.